pkg/runtime/runtime.go

```diff
@@ -408,6 +408,10 @@ func (a *DaprRuntime) setupTracing(hostAddress string, tpStore tracerProviderSto
 		tpStore.RegisterExporter(otelExporter)
 	}
 
+	if !tpStore.HasExporter() && tracingSpec.SamplingRate != "" {
```

Comment:
Note that this entire code (and the code above) won't be used if Sampling Rate is 0 anyway